### PR TITLE
Update even more Core Validation functions to pre/postcall format

### PR DESCRIPTION
**Pull request by shannon-lunarg**
More work done towards converting Core Validation functions to a precall/postcall format.
This PR addresses functions with the word "Reset" in their name.

Batch 3 of many.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5760,25 +5760,35 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateDescriptorPool(VkDevice device, const VkDes
     return result;
 }
 
+static bool PreCallValidateResetDescriptorPool(layer_data *dev_data, VkDescriptorPool descriptorPool) {
+    return ValidateIdleDescriptorSetForPoolReset(dev_data, descriptorPool);
```
Comment (mikes-lunarg) [inline layers/core_validation.cpp]: This function ends up being a really simple wrapper, you could just rename `ValidateIdleDescriptorSetForPoolReset` to `PreCallValidateResetDescriptorPool` and save a function call

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -5760,25 +5760,35 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateDescriptorPool(VkDevice device, const VkDes
...
+}
+
+static void PostCallRecordResetDescriptorool(layer_data *dev_data, VkDevice device, VkDescriptorPool descriptorPool,
```
Comment (mikes-lunarg) [inline layers/core_validation.cpp]: function name typo

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -5760,25 +5760,35 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateDescriptorPool(VkDevice device, const VkDes
...
+                                             VkDescriptorPoolResetFlags flags) {
+    ClearDescriptorPool(dev_data, device, descriptorPool, flags);
```
Comment (mikes-lunarg) [inline layers/core_validation.cpp]: Same comment as on `PreCallValidateResetDescriptorPool`, If `ClearDescriptorPool` does everything you need you can just rename it to `PostCallRecordResetDescriptorPool`

**Comment by shannon-lunarg** [comment]
@mikes-lunarg The requested changes have been made.

**Pull request merged**